content/protobuf/language-mapping/service.md

```diff
@@ -0,0 +1,220 @@
+[address](/icerpc/invocation/service-address) of the remote service, and the `encodeOptions` parameter allows
+you to customize the encoding of your Protobuf messages. See [ProtobufEncodeOptions] for details.
+
+A `null` service address is equivalent to an icerpc service address with the default path of the associated Protobuf
+service.
+
+{% callout type="note" %}
+The default path of a Protobuf service is `/` followed by its fully qualified name. For example, the default path of
+Protobuf service `visitor_center.Greeter` is `/visitor_center.Greeter`.
+{% /callout %}
+
+### I*Name*Service
+
+Interface I*Name*Service is a server-side helper: it helps you create a service (a C# class) that implements Protobuf
+service _Name_.
+
+The principle is straightforward: your service class must be a partial class that implements I*Name*Service. It must
+also carry the [ProtobufService] attribute.
+
+The `ProtobufService` attribute instructs the Protobuf Service source generator to implement interface [IDispatcher] by
+directing incoming requests to I*Name*Service methods based on the Protobuf rpc method names.
+
+For example:
+
+{% side-by-side alignment="top" %}
+
+```protobuf
+syntax = "proto3";
+package example;
+option csharp_namespace = "Example";
+
+service Widget {
+    rpc Spin (SpinConfig)
+        returns (google.protobuf.Empty);
+}
+```
+
+```csharp
+namespace Example;
+
+// Generated code
+public partial interface IWidgetService
+{
+    // One method per operation
+    ValueTask<WellKnownTypes.Empty> SpinAsync(
+        SpinConfig message,
+        IFeatureCollection features,
+        CancellationToken cancellationToken);
+}
+
+// Application code
+[ProtobufService]
+internal partial class MyWidget : IWidgetService
+{
+    // implement SpinAsync ...
+}
+```
+
+{% /side-by-side %}
+
+{% callout type="note" %}
+Even though I*Name*Service is an interface, it's not used as an abstraction: you shouldn't make calls to this interface
+or create decorators for this interface. It's just a model that your service class must implement.
+{% /callout %}
+
+Note that the same service class can implement any number of Protobuf services provided their rpc methods have unique
+names. For example:
+
+{% side-by-side alignment="top" %}
+
+```protobuf
+syntax = "proto3";
+package example;
+option csharp_namespace = "Example";
+
+service Widget {
+    rpc Spin (SpinConfig)
+        returns (google.protobuf.Empty);
+}
+
+service Counter {
+    rpc GetCount (google.protobuf.Empty)
+        returns (Count);
+}
+```
+
+```csharp
+// A service class that implements 2 Protobuf
+// services
+[ProtobufService]
+internal partial class MyWidget : IWidgetService,
+                                  ICounterService
+{
+    // implements SpinAsync and GetCountAsync.
+}
+```
+
+{% /side-by-side %}
+
+### deprecated option
+
+If you set option `deprecated` to `true` on a Protobuf service, the 2 interfaces and struct generated by
```

Comment:
This might be over the 120 column limit with this change.
```suggestion
If you set the `deprecated` option to `true` on a Protobuf service, the 2 interfaces and the struct generated by
```